src/utils/api.js

```diff
@@ -20,14 +39,49 @@ let newListing = {
   condition: "",
   location: "",
   tradeRequired: false,
-  user_id: ""
+  user_id: "",
 };
 
+let listingRef;
 export const postListing = async (event, newListing) => {
   event.preventDefault();
-
   const docRef = await addDoc(collection(db, "listings"), newListing);
   console.log("Document written to Listings with ID: ", docRef.id);
+  listingRef = docRef.id;
+  uploadImage(event);
+  return listingRef;
+};
+
+const storageRef = ref_storage(storage);
```

Comment:
does this need defining outside of the uploadImage function if its being re-assigned in it and only used within that function? Looks as though you can just have it defined inside and passed on as needed.